SCons/Tool/MSCommon/MSVC/Util.py

```diff
@@ -26,16 +26,25 @@
 """
 
 import os
+import pathlib
 import re
 
 from collections import (
     namedtuple,
 )
 
+from ..common import debug
+
 from . import Config
 
 # path utilities
 
+# windows drive specification (e.g., 'C:')
+_RE_DRIVESPEC = re.compile(r'^[A-Za-z][:]$', re.IGNORECASE)
+
+# windows path separators
+_OS_PATH_SEPS = ('\\', '/')
```

Comment:
Could you use os.sep and os.altsep here instead?

Reply:
Possibly, but it would have to guaranteed to be `windows` or `win32`.

Reply:
> Possibly, but it would have to guaranteed to be `windows` or `win32`.

The assumption you're on win32 if using msvc is pretty baked in - and not entirely correct, since you can get VS on the Mac now.

Reply:
True.

Unfortunately, there exist "public" methods that could be called from site initialization and/or sconstruct files that could bypass the regular tool loading and make use of the internal MSVC utility functions.

Might need some form of protection for the "public" functions.

Reply:
Not necessary. If people want to do truly silly things. We don't need to guard against that and waste our time...

Reply:
It might just work.

The resolve_path and normalize_path functions are exercised somewhat in the MSCommon/MSVC/UtilTests.py script on linux in the test suite (and occasionally locally in WSL) which does pass on non-windows platforms.

However, testing resolve_path handling of drive specifications (i.e., "C:") is only done on windows.

Reply:
In general if we have to jump through too many hoops we've had no issues with dropping some ability to test platform stuff on other platforms.   Especially given github actions, appveyor, etc.

In the "old" days, we had to run our own buildbot and depend on our own and other users volunteered build workers.  I'm glad we've moved past that as it was a major time sink..